### PR TITLE
docs: fix spelling error

### PR DESCRIPTION
**Pull request by Gaiejj**
## Description

docs: fix spelling error

## Motivation and Context

Why is this change required? What problem does it solve?
If it fixes an open issue, please link to the issue here.
You can use the syntax `close #15213` if this solves the issue #15213

- [ ] I have raised an issue to propose this change ([required](https://github.com/PKU-MARL/omnisafe/issues) for new features and bug fixes)

## Types of changes

What types of changes does your code introduce? Put an `x` in all the boxes that apply:

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds core functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Documentation (update in the documentation)

## Checklist

Go over all the following points, and put an `x` in all the boxes that apply.
If you are unsure about any of these, don't hesitate to ask. We are here to help!

- [x] I have read the [CONTRIBUTION](https://omnisafe.readthedocs.io/en/latest/developer/contributing.html) guide. (**required**)
- [x] My change requires a change to the documentation.
- [x] I have updated the tests accordingly. (*required for a bug fix or a new feature*)
- [x] I have updated the documentation accordingly.
- [x] I have reformatted the code using `make format`. (**required**)
- [x] I have checked the code using `make lint`. (**required**)
- [x] I have ensured `make test` pass. (**required**)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -22,29 +22,31 @@ The simulation environment around OmniSafe and a series of reliable algorithm im
...
+  - [Installation](#installation)
+    - [Prerequisites](#prerequisites)
+    - [Install from source](#install-from-source)
+    - [Examples](#examples)
+  - [Getting Started](#getting-started)
+    - [1. Run Agent from preset yaml file](#1-run-agent-from-preset-yaml-file)
+    - [2. Run Agent from custom config dict](#2-run-agent-from-custom-config-dict)
+    - [3. Run Agent from custom terminal config](#3-run-agent-from-custom-terminal-config)
+  - [The OmniSafe Team](#the-omnisafe-team)
+  - [License](#license)
```
Comment (XuehaiPan) [inline README.md:50]: Install VS Code extension _Markdown All in One_ and regenerate this TOC.

File: docs/source/conf.py [hunk 2]
```diff
@@ -34,28 +29,8 @@
...
-    'sphinx.ext.viewcode',
-    'sphinx.ext.githubpages',
-    'sphinx.ext.extlinks',
-    'recommonmark',
-    'sphinx_markdown_tables',
-    'sphinx_design',
-    'sphinx_copybutton',
-    'sphinx_autodoc_typehints',
-]
+extensions = ['recommonmark', 'sphinx_markdown_tables', 'sphinx_design', 'sphinx_copybutton']
```
Comment (XuehaiPan) [inline docs/source/conf.py:32]: Revert this file.

File: docs/source/spelling_wordlist.txt [hunk 3]
```diff
@@ -205,7 +206,9 @@ Peng
 Jiadong
 Guo
 Bo
+Dai
 lei
+bool
```
Comment (XuehaiPan) [inline docs/source/spelling_wordlist.txt:211]: Revert changes in this file above.

File: docs/source/spelling_wordlist.txt [hunk 4]
```diff
@@ -223,3 +226,46 @@ Nachum
 Aleksandra
 Duenez
 Ghavamzadeh
+Karush
+Levin
+
+optimality
+invertible
+neighbourhood
```
Comment (XuehaiPan) [inline docs/source/spelling_wordlist.txt:234]: Use American English to fit the en-US dictionary.

**Review by XuehaiPan — CHANGES_REQUESTED** [review]
Enable spelling check in the `lint` target in `Makefile` and enable `make spelling` in GitHub Action workflows.

**Review by XuehaiPan — APPROVED** [review]

**Review by zmsn-2077 — APPROVED** [review]

**Pull request merged**